bin/sozo/src/commands/auth.rs

```diff
@@ -158,22 +83,47 @@ pub enum AuthCommand {
 
 impl AuthArgs {
     pub fn run(self, config: &Config) -> Result<()> {
-        let env_metadata = if config.manifest_path().exists() {
-            let ws = scarb::ops::read_workspace(config.manifest_path(), config)?;
+        let env_metadata = utils::load_metadata_from_config(config)?;
 
-            dojo_metadata_from_workspace(&ws).and_then(|inner| inner.env().cloned())
-        } else {
-            None
+        let (kind, world, starknet, account, transaction) = match self.command {
+            AuthCommand::Grant { kind, world, starknet, account, transaction } => {
+                (kind, world, starknet, account, transaction)
+            }
+            AuthCommand::Revoke { kind, world, starknet, account, transaction } => {
+                (kind, world, starknet, account, transaction)
+            }
```

Comment:
Here, we have the match to destructure the command, but we lose the actual information if we are granting or revoking.
The code should be adjusted to ensure we execute the correct command (even if for now revoke is not being used).